galpy/df/streamspraydf.py

```diff
@@ -355,3 +296,366 @@ def _setup_rot(self, dt):
+                    [0, 0, 144, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0],
+                    [-4.9, 0, 0, 0, 400, 0],
+                    [0, 0, 0, 0, 0, 484],
+                ]
+            )
+        else:
+            self._cov = cov
+        return None
+
+    def spray_df(self, xyzpt, vxyzpt, dt):
+        """
+        Sample the positions and velocities around the progenitor
+
+        Parameters
+        ----------
+        xyzpt : array, shape (N,3)
+            Positions of progenitor in the progenitor coordinates.
+        vxyzpt : array, shape (N,3)
+            Velocities of progenitor in the progenitor coordinates.
+        dt : array, shape (N,)
+            Time of sampling.
+
+        Returns
+        -------
+        xst, yst, zst : array, shape (N,)
+            Positions of points on the stream in the progenitor coordinates.
+        vxst, vyst, vzst : array, shape (N,)
+            Velocities of points on the stream in the progenitor coordinates.
+        """
+        Rpt, phipt, Zpt = coords.rect_to_cyl(xyzpt[:, 0], xyzpt[:, 1], xyzpt[:, 2])
+        vRpt, vTpt, vZpt = coords.rect_to_cyl_vec(
+            vxyzpt[:, 0], vxyzpt[:, 1], vxyzpt[:, 2], Rpt, phipt, Zpt, cyl=True
+        )
+        try:
+            rtides = rtide(
+                self._rtpot,
+                Rpt,
+                Zpt,
+                phi=phipt,
+                t=-dt,
+                M=self._progenitor_mass,
+                use_physical=False,
+            )
+        except (ValueError, TypeError):
+            rtides = numpy.array(
+                [
+                    rtide(
+                        self._rtpot,
+                        Rpt[ii],
+                        Zpt[ii],
+                        phi=phipt[ii],
+                        t=-dt[ii],
+                        M=self._progenitor_mass,
+                        use_physical=False,
+                    )
+                    for ii in range(len(Rpt))
+                ]
+            )
+
+        # Sample positions and velocities in the instantaneous frame
+        posvel = numpy.random.multivariate_normal(self._mean, self._cov, size=len(dt))
+        Dr = posvel[:, 0] * rtides
+        v_esc = numpy.sqrt(2 * self._progenitor_mass / Dr)
+        Dv = posvel[:, 3] * v_esc
+        if self._leading:
+            Dr *= -1.0
+            Dv *= -1.0
+
+        # convert degrees to radians
+        phi = posvel[:, 1] * numpy.pi / 180
+        theta = posvel[:, 2] * numpy.pi / 180
+        alpha = posvel[:, 4] * numpy.pi / 180
+        beta = posvel[:, 5] * numpy.pi / 180
+
+        xst = xyzpt[:, 0] + Dr * numpy.cos(theta) * numpy.cos(phi)
+        yst = xyzpt[:, 1] + Dr * numpy.cos(theta) * numpy.sin(phi)
+        zst = xyzpt[:, 2] + Dr * numpy.sin(theta)
+        vxst = vxyzpt[:, 0] + Dv * numpy.cos(beta) * numpy.cos(alpha)
+        vyst = vxyzpt[:, 1] + Dv * numpy.cos(beta) * numpy.sin(alpha)
+        vzst = vxyzpt[:, 2] + Dv * numpy.sin(beta)
```

Comment:
Use ``util.coords.spher_to_cyl(_vec)`` and ``util.coords.cyl_to_rect(_vec)`` here.